sound/soc/sof/pm.c

```diff
@@ -265,14 +265,17 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 			ret);
 
 	/* Do not reset FW state if DSP is in D0 */
-	if (target_state == SOF_DSP_PM_D0)
-		return ret;
+	if (ret == 0 && target_state == SOF_DSP_PM_D0)
+		return 0;
 
 	/* reset FW state */
 	sof_set_fw_state(sdev, SOF_FW_BOOT_NOT_STARTED);
```

Comment:
If the suspend callback for D0ix failed, the FW state would be incorrect isnt it?